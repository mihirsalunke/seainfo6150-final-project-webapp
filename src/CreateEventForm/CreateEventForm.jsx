import React, { useState } from "react";
import AddEventConfirmation from "../AddEventConfirmation/AddEventConfirmation.jsx";
import styles from "./CreateEventForm.module.css";

const CreateEventForm = () => {
    const [submittedForm, setSubmittedForm] = useState();
    function onSubmit(e) {
        e.preventDefault();
        const data = new FormData(e.target);
        setSubmittedForm(data);
    };

    return (
        <div>
            {
                submittedForm ? (
                    <>
                        <AddEventConfirmation eventName={submittedForm.get("myEventName")} eventDate={submittedForm.get("myEventDate")} eventTime={submittedForm.get("myEventTime")} eventLocation={submittedForm.get("myLocation")} />
                    </>
                ) : (
                    <>
                        <div className={styles.container}>
                            <div className={styles.title}>
                                <p>Create New Event:</p>
                            </div>
                            <form onSubmit={onSubmit} className={styles.createEventForm}>
                                <div className={styles.formField}>
                                    <label htmlFor="organizerNameId">Organizer Name: </label>
                                    <input type="text" name="myOrganizerName" id="organizerNameId" required/>
                                </div>
                                
                                <div className={styles.formField}>
                                    <label htmlFor="contactId">Phone Number: </label>
                                    <input type="tel" name="myContactNumber" id="contactId" required/>
                                </div>
                                
                                <div className={styles.formField}>
                                    <label htmlFor="eventId">Event Name: </label>
                                    <input type="text" name="myEventName" id="eventId" required/>
                                </div>
                                
                                <div className={styles.formField}>
                                    <label htmlFor="eventDate">Event Date: </label>
                                    <input type="date" name="myEventDate" id="eventDate" required/>
                                </div>
                                
                                <div className={styles.formField}>
                                    <label htmlFor="eventTime">Event Time: </label>
                                    <input type="time" name="myEventTime" id="eventTime" required/>
                                </div>
                                
                                <div className={styles.formField}>
                                    <label htmlFor="location">Location: </label>
                                    <input type="text" name="myLocation" id="location" required/>
                                </div>
                                
                                <div className={styles.formDropdown}>
                                    <label htmlFor="categoryDropdown">Event Category: </label>
                                    <select name="myCategoryDropdown" id="categoryDropdown">
                                        <option value="Sports">Sports</option>
                                        <option value="Tech">Tech</option>
                                        <option value="Music">Music</option>
                                        <option value="Fashion">Fashion</option>
                                        <option value="Health">Health</option>
                                        <option value="Other">Other</option>
                                    </select>
                                </div>
                                
                                <div className={styles.formRadio}>
                                    <p>Event Type:</p>
                                    <div className={styles.radio}>
                                        <input type="radio" name="eventType" id="publicEvent" value="public"/>
                                        <label htmlFor="publicEvent">Public</label>
                                    </div>
                                    <div className={styles.radio}>
                                        <input type="radio" name="eventType" id="privateEvent" value="private"/>
                                        <label htmlFor="privateEvent">Private</label>
                                    </div>
                                </div>
                                
                                <div className={styles.formCheckbox}>
                                    <div className={styles.checkbox}>
                                        <input type="checkbox" name="eventOption" id="inviteOnly" value="invite required"/>
                                        <label htmlFor="inviteOnly">Make the event invite only</label>
                                    </div>
                                    <div className={styles.checkbox}>
                                        <input type="checkbox" name="eventOption" id="idRequired" value="id required"/>
                                        <label htmlFor="idRequired">Participant's ID required</label>
                                    </div>
                                </div>

                                <div className={styles.buttonContainer}>
                                    <input className={styles.submitButton} type="submit" value="Create Event"/>
                                </div>
                            </form>
                        </div>
                    </>
                )
            }
        </div>
    );
};

export default CreateEventForm;