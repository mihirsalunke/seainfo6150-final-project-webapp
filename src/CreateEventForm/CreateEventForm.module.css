@import url("https://fonts.googleapis.com/css2?family=Rowdies:wght@300&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Nerko+One&display=swap");

.container {
    background-image: linear-gradient(to right, rgb(160, 201, 226), rgb(207, 233, 248) 70%);
    max-width: 700px;
    margin: 2rem auto;
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.3);
    justify-content: center;
    border-radius: 10px;
}

.title {
    text-align: center;
    font-family: Rowdies, sans-serif;
    font-size: 2em;
}

.createEventForm {
    margin: 2em 5em;
    display: flex;
    flex-flow: column wrap;
    font-size: 100%;
    color:  black;
}

.formField {
    display: flex;
    padding: 1%;
}
.formField label {
    width: 50%;
    margin-right: 2%;
    text-align: right;
    font-family: Rowdies, sans-serif;
}
.formField input {
    width: 50%;
    margin-left: 2%;
}


.formDropdown {
    display: flex;
    padding: 2%;
    align-items: center;
}
.formDropdown label {
    width: 50%;
    margin-right: 2%;
    text-align: right;
    font-family: Rowdies, sans-serif;
}
.formDropdown select {
    width: 50%;
    margin-left: 2%;
}

.formRadio {
    display: flex;
    padding: 2%;
    align-items: center;
}
.formRadio p {
    font-family: Rowdies, sans-serif;
}
.radio {
    display: flex;
}

.checkbox {
    padding: 1%;
}

.buttonContainer {
    margin-top: 1em;
    align-self: center;
}

.submitButton {
    background: orange;
    border-radius: 10px;
    color: #fff;
    padding: 10px;
    font-size: 1.5em;
    font-family: Rowdies, sans-serif;
    margin: 0 0 20px;
    width: 7em;
    text-align: center;
    vertical-align: middle;

    transition: transform .25s ease-out;
}

.submitButton:hover {
    background: green;
    transform: translateY(2px);
    cursor: pointer;
}

.submitButton:active {
    transform: translateY(1px);
}