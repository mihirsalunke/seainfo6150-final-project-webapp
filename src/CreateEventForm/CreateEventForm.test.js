import React from  "React";
import CreateEventForm from "./CreateEventForm.jsx";
import { render } from "@testing-library/react";

describe("CreateEventForm tests", () => {
    it("renders correctly", () => {
      const { container } = render(<CreateEventForm />);
      expect(container).toMatchSnapshot();
    });
});