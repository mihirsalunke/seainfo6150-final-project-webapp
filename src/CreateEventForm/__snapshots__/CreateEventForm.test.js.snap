// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEventForm tests renders correctly 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <div
        class="title"
      >
        <p>
          Create New Event:
        </p>
      </div>
      <form
        class="createEventForm"
      >
        <div
          class="formField"
        >
          <label
            for="organizerNameId"
          >
            Organizer Name: 
          </label>
          <input
            id="organizerNameId"
            name="myOrganizerName"
            required=""
            type="text"
          />
        </div>
        <div
          class="formField"
        >
          <label
            for="contactId"
          >
            Phone Number: 
          </label>
          <input
            id="contactId"
            name="myContactNumber"
            required=""
            type="tel"
          />
        </div>
        <div
          class="formField"
        >
          <label
            for="eventId"
          >
            Event Name: 
          </label>
          <input
            id="eventId"
            name="myEventName"
            required=""
            type="text"
          />
        </div>
        <div
          class="formField"
        >
          <label
            for="eventDate"
          >
            Event Date: 
          </label>
          <input
            id="eventDate"
            name="myEventDate"
            required=""
            type="date"
          />
        </div>
        <div
          class="formField"
        >
          <label
            for="eventTime"
          >
            Event Time: 
          </label>
          <input
            id="eventTime"
            name="myEventTime"
            required=""
            type="time"
          />
        </div>
        <div
          class="formField"
        >
          <label
            for="location"
          >
            Location: 
          </label>
          <input
            id="location"
            name="myLocation"
            required=""
            type="text"
          />
        </div>
        <div
          class="formDropdown"
        >
          <label
            for="categoryDropdown"
          >
            Event Category: 
          </label>
          <select
            id="categoryDropdown"
            name="myCategoryDropdown"
          >
            <option
              value="Sports"
            >
              Sports
            </option>
            <option
              value="Tech"
            >
              Tech
            </option>
            <option
              value="Music"
            >
              Music
            </option>
            <option
              value="Fashion"
            >
              Fashion
            </option>
            <option
              value="Health"
            >
              Health
            </option>
            <option
              value="Other"
            >
              Other
            </option>
          </select>
        </div>
        <div
          class="formRadio"
        >
          <p>
            Event Type:
          </p>
          <div
            class="radio"
          >
            <input
              id="publicEvent"
              name="eventType"
              type="radio"
              value="public"
            />
            <label
              for="publicEvent"
            >
              Public
            </label>
          </div>
          <div
            class="radio"
          >
            <input
              id="privateEvent"
              name="eventType"
              type="radio"
              value="private"
            />
            <label
              for="privateEvent"
            >
              Private
            </label>
          </div>
        </div>
        <div
          class="formCheckbox"
        >
          <div
            class="checkbox"
          >
            <input
              id="inviteOnly"
              name="eventOption"
              type="checkbox"
              value="invite required"
            />
            <label
              for="inviteOnly"
            >
              Make the event invite only
            </label>
          </div>
          <div
            class="checkbox"
          >
            <input
              id="idRequired"
              name="eventOption"
              type="checkbox"
              value="id required"
            />
            <label
              for="idRequired"
            >
              Participant's ID required
            </label>
          </div>
        </div>
        <div
          class="buttonContainer"
        >
          <input
            class="submitButton"
            type="submit"
            value="Create Event"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
